chain/x/zoracle/internal/keeper/report.go

```diff
@@ -3,27 +3,31 @@ package keeper
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
-	"github.com/bandprotocol/d3n/chain/wasm"
 	"github.com/bandprotocol/d3n/chain/x/zoracle/internal/types"
 )
 
 // SetReport is a function that saves a data report to storeage.
-func (k Keeper) SetReport(ctx sdk.Context, requestID uint64, validatorAddress sdk.ValAddress, data []byte) {
+func (k Keeper) SetReport(
+	ctx sdk.Context,
+	requestID int64,
+	validatorAddress sdk.ValAddress,
+	data []types.ExternalData,
+) {
 	key := types.ReportStoreKey(requestID, validatorAddress)
-	report := types.NewReport(data, uint64(ctx.BlockHeight()))
+	report := types.NewReport(data, ctx.BlockHeight())
 	store := ctx.KVStore(k.storeKey)
 	store.Set(key, k.cdc.MustMarshalBinaryBare(report))
 }
 
 // GetReportsIterator returns an iterator for all reports for a specific request ID.
-func (k Keeper) GetReportsIterator(ctx sdk.Context, requestID uint64) sdk.Iterator {
+func (k Keeper) GetReportsIterator(ctx sdk.Context, requestID int64) sdk.Iterator {
 	prefix := types.GetIteratorPrefix(types.ReportKeyPrefix, requestID)
 	store := ctx.KVStore(k.storeKey)
 	return sdk.KVStorePrefixIterator(store, prefix)
 }
 
 // GetDataReports returns all the reports for a specific request ID.
-func (k Keeper) GetDataReports(ctx sdk.Context, requestID uint64) []types.Report {
+func (k Keeper) GetDataReports(ctx sdk.Context, requestID int64) []types.Report {
```

Comment:
Some of these might not be needed anymore. We will see after we finish the handler (: